### PR TITLE
Update PR body when it already exists

### LINKED ISSUE #108: Update commit hash when register() is called on a new commit
See https://github.com/christopher-dG/tag-bot/issues/3 for tag-bot issue.

Because the commit hash is not updated when register() is called to fix a commit, tag-bot does not tag the correct commit.

### PR DESCRIPTION
**Pull request by christopher-dG**
Fixes #108 
Fixes https://github.com/christopher-dG/tag-bot/issues/3 (am I allowed to do this?!)

I haven't tested this, maybe someone more familiar with test-deploying Registrator and using GitHub.jl can tell me if this is going to break or not.

### CODE REVIEW

File: src/Server.jl [hunk 1]
```diff
@@ -743,6 +743,8 @@ function make_pull_request(pp::ProcessedParams, rp::RequestParams, rbrn::RegBran
     catch ex
         if is_pr_exists_exception(ex)
             @debug("Pull request already exists, not creating")
+            params = Dict("body" => params["body"])
```
Comment (fredrikekre) [inline src/Server.jl]: You need more params here, in particular the PR number since the call on the next line wouldn't know what PR to update.

Comment (christopher-dG) [reply src/Server.jl]: Oh right, of course.

Comment (christopher-dG) [reply src/Server.jl]: I think this might do the right thing now

File: src/Server.jl [hunk 2]
```diff
@@ -736,13 +736,24 @@ function make_pull_request(pp::ProcessedParams, rp::RequestParams, rbrn::RegBran
...
     try
-        pr = create_pull_request(repo; auth=get_user_auth(), params=params)
+        pr = create_pull_request(repo; auth=auth, params=params)
         msg = "created"
         @debug("Pull request created")
     catch ex
         if is_pr_exists_exception(ex)
-            @debug("Pull request already exists, not creating")
+            @debug("Pull request already exists, trying to update body")
+            params = Dict(
```
Comment (fredrikekre) [inline src/Server.jl]: Do you need to create this? Can't you just use the one from L715?

File: src/Server.jl [hunk 3]
```diff
@@ -736,13 +736,24 @@ function make_pull_request(pp::ProcessedParams, rp::RequestParams, rbrn::RegBran
...
+                "base" => target_registry["base_branch"],
+                "head" => split(repo, "/")[1] * ":" * brn,
+            )
+            prs = pull_request(repo; auth=auth, params=params)
+            if !isempty(prs)
+                pr = prs[1]
+                params = Dict("body" => params["body"])
```
Comment (fredrikekre) [inline src/Server.jl]: You have overwritten `params` here on L747 so the body does not exist, right?

File: src/Server.jl (hunk 2 above)
Comment (christopher-dG) [reply src/Server.jl]: Well I don't want to send any unwanted values that may or may not affect the API call.

File: src/Server.jl (hunk 3 above)
Comment (christopher-dG) [reply src/Server.jl]: Right, that's true

**Comment by christopher-dG** [comment]
I've dealt with the overwritten variable by just constructing the parameters inline.

File: src/Server.jl [hunk 4]
```diff
@@ -736,13 +736,22 @@ function make_pull_request(pp::ProcessedParams, rp::RequestParams, rbrn::RegBran
...
-        pr = create_pull_request(repo; auth=get_user_auth(), params=params)
+        pr = create_pull_request(repo; auth=auth, params=params)
         msg = "created"
         @debug("Pull request created")
     catch ex
         if is_pr_exists_exception(ex)
-            @debug("Pull request already exists, not creating")
+            @debug("Pull request already exists, trying to update body")
+            prs = pull_request(repo; auth=auth, params=Dict(
+                "base" => target_registry["base_branch"],
```
Comment (fredrikekre) [inline src/Server.jl]:
```suggestion
                "base" => params["base"],
```

File: src/Server.jl [hunk 5]
```diff
@@ -736,13 +736,22 @@ function make_pull_request(pp::ProcessedParams, rp::RequestParams, rbrn::RegBran
...
+                "head" => split(repo, "/")[1] * ":" * brn,
```
Comment (fredrikekre) [inline src/Server.jl]:
```suggestion
                "head" => params["head"],
```

Comment (christopher-dG) [reply src/Server.jl]: I don't think this does what we want, the GitHub docs seem to say that the value must also contain the user. https://developer.github.com/v3/pulls/#parameters

**Pull request merged**